api/app/hfi/hfi_calc.py

```diff
@@ -308,18 +308,20 @@ def calculate_daily_results(num_prep_days: int,
                             area_dailies: List[StationDaily],
                             planning_area_fire_starts: Dict[int, FireStartRange],
                             area_id: int,
-                            fire_start_lookup: Dict[int, Dict[int, int]]) -> Tuple[List[DailyResult], bool]:
+                            fire_start_lookup: Dict[int, Dict[int, int]],
+                            num_unique_station_codes: int) -> Tuple[List[DailyResult], bool]:
     """ Calculate the daily results for a planning area."""
     daily_results: List[DailyResult] = []
     for index in range(num_prep_days):
         dailies_date = start_date + timedelta(days=index)
         prep_day_dailies = get_prep_day_dailies(dailies_date, area_dailies)
         daily_fire_starts: FireStartRange = planning_area_fire_starts[area_id][index]
-        mean_intensity_group = calculate_mean_intensity(prep_day_dailies)
+        mean_intensity_group = calculate_mean_intensity(prep_day_dailies, num_unique_station_codes)
         prep_level = calculate_prep_level(mean_intensity_group, daily_fire_starts, fire_start_lookup)
         validated_dailies: List[ValidatedStationDaily] = list(map(validate_station_daily, prep_day_dailies))
         # check if all validated_dailies are valid.
-        all_dailies_valid = all(map(lambda validated_daily: (validated_daily.valid), validated_dailies))
```

Comment:
something about this wasn't working - it kept marking everything as valid always.

Reply:
Weird, maybe the `map` needed to be wrapped in a `list`? Perhaps `all` doesn't know how to handle a wrap `map` iterable object.